src/App.js

```diff
@@ -10,568 +10,687 @@ import CodeMirror from '@uiw/react-codemirror';
-                                          genHtml(elems) + '\n  </div>\n</div>'}
-                                  extensions={[markdown({ base: markdownLanguage, codeLanguages: languages })]} 
-                      />
-                    </div></div>
                 </div>
-              </div>
             </div>
-
-          </div>
-          <div className='col-md-6 col-sm-6 output_form maxht' onDragEnter={(e) => dragPosition.current = 1}>
-            {(elems.length < 1) && <div
-              className='empty'
-              onDragEnter={(e) => { if (!$(e.target).hasClass('hover')) $(e.target).addClass('hover') }}
-              onDragLeave={(e) => { if ($(e.target).hasClass('hover')) $(e.target).removeClass('hover') }}
-            >
-              Add elements to your form</div>}
-            {
-
-              elems && elems.map(function (e, i) {
-                return (
-                  <OverlayTrigger
-                    container={this}
-                    trigger="click"
-                    rootClose
-                    placement="left"
-                    overlay={popoverLeft}
-                    key={i}
-                  >
-                    <div
-                      className="input_elem indragitem"
-                      draggable="true"
-                      onClick={(e) => handleDetails(e, i)}
-                      onDragStart={(e) => dragItem.current = i}
-                      onDragEnter={(e) => DragOverEl(e, i)}
-                      onDragLeave={(e) => DragLeaveEl(e, i)}
-                      onDragEnd={(e) => arrangeElem(e)}
-                      onDragOver={(e) => e.preventDefault()}
-                    >
-                      {Parser(e)}
-                    </div>
-                  </OverlayTrigger>
-                )
-              })
-            }
-          </div>
-        </div>
-      </div>
-    </>
-  );
+        </>
+    );
 }
 
+let checkboxHTML = '<div class="row">' +
+    '  <div class="form-check col-xs-12 pull-left input-group">' +
+    '    <input class="form-check-input form-control" type="checkbox" name="cp-checkbox-element-name" id="cp-checkbox-element-id" value="VALUE"/>' +
+    '    <label class="cp_label col-xs-10 pull-right" for="cp-checkbox-element-for">VALUE</label>' +
+    '  </div>' +
+    '</div>';
+
+let radioButtonHTML = '<div class="row">' +
+    '  <div class="form-check input-group">' +
+    '    <input class="form-check-input col-xs-1" type="radio" name="cp-radio-button-element-name" id="cp-radio-button-element-id" value="VALUE">' +
+    '    <label class="form-check-label col-xs-10" for=cp-radio-button-element-for"> VALUE</label>' +
+    '  </div>' +
+    '</div>';
+
+
```

Comment:
nb: (for my own understanding) why were these taken out?

Reply:
Because these HTML code blocks were used in two different places and I should have modified them both. It seemed reasonable to me to put them into variables instead. I think they can be either inside or outside the App function (it's debatable and I'm not entirely sure which would be the better approach), but I definitely should have used  `const` instead of `let` since they will not change. I will fix it, thanks.